packages/devextreme/js/__internal/scheduler/appointments/m_appointment_collection.ts

```diff
@@ -804,23 +804,30 @@ class SchedulerAppointments extends CollectionWidget {
   _correctEndDateByDelta(endDate, deltaTime) {
     const endDayHour = this.invoke('getEndDayHour');
     const startDayHour = this.invoke('getStartDayHour');
-    let result = endDate.getTime() + deltaTime;
-    const visibleDayDuration = (endDayHour - startDayHour) * toMs('hour');
-
-    const daysCount = deltaTime > 0
-      ? Math.ceil(deltaTime / visibleDayDuration)
-      : Math.floor(deltaTime / visibleDayDuration);
 
     const maxDate = new Date(endDate);
     const minDate = new Date(endDate);
 
     minDate.setHours(startDayHour, 0, 0, 0);
     maxDate.setHours(endDayHour, 0, 0, 0);
 
+    const correctEndDate = new Date(endDate);
+
+    if (correctEndDate.getTime() > maxDate.getTime()) {
+      correctEndDate.setHours(endDayHour, 0, 0, 0);
+    }
```

Comment:
Minor comment.

Let's reorganize this code a little:
```suggestion
    const maxDate = new Date(endDate);
    const minDate = new Date(endDate);
    const correctEndDate = new Date(endDate);

    minDate.setHours(startDayHour, 0, 0, 0);
    maxDate.setHours(endDayHour, 0, 0, 0);

    if (correctEndDate > maxDate) {
      correctEndDate.setHours(endDayHour, 0, 0, 0);
    }
```

And we can compare date variables without calling ```getTime()``` because under the hood comparison is made by comparing ms date values :)